src/blockdigest.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "blockdigest.h"
+#include "streams.h"
+#include "version.h"
+#include "uint256.h"
+#include "primitives/transaction.h"
+#include "primitives/block.h"
+#include "validation.h"
+#include "chainparams.h"
+#include "utiltime.h"
+
+namespace DigestFilter {
+
+data_t base_filter::outpoint_data(const COutPoint& outpoint) const
+{
+    CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
+    stream << outpoint;
+    data_t data(stream.begin(), stream.end());
+    return data;
+}
+
+data_t base_filter::hash_data(const uint256& hash) const
+{
+    data_t data(hash.begin(), hash.end());
+    return data;
+}
+
+std::vector<const data_t> base_filter::tx_data_vec(const CTransaction& tx) const
+{
+    std::vector<const data_t> tx_data;
+    const uint256& hash = tx.GetHash();
+    tx_data.push_back(hash_data(hash));
+
+    for (unsigned int i = 0; i < tx.vout.size(); i++) {
+        const CTxOut& txout = tx.vout[i];
+        CScript::const_iterator pc = txout.scriptPubKey.begin();
+        data_t data;
+        while (pc < txout.scriptPubKey.end()) {
```

Comment:
TBH this is copy pasted from the bloom filter code. I just presumed it would do the right thing. I.e. "I don't know but I'll investigate!" Thanks for the pointer. :)